src/ImageSharp.Drawing/Paths/InternalPath.cs

```diff
@@ -164,6 +169,45 @@ public PointInfo DistanceFromPath(Vector2 point)
         }
 
         /// <summary>
+        /// Finds the intersections.
+        /// </summary>
+        /// <param name="start">The start.</param>
+        /// <param name="end">The end.</param>
+        /// <param name="buffer">The buffer.</param>
+        /// <param name="count">The count.</param>
+        /// <param name="offset">The offset.</param>
+        /// <returns>number iof intersections hit</returns>
+        public int FindIntersections(Vector2 start, Vector2 end, Vector2[] buffer, int count, int offset)
+        {
+            int polyCorners = this.points.Length;
+
+            if (!this.closedPath)
+            {
+                polyCorners -= 1;
+            }
+
+            int position = 0;
+            for (int i = 0; i < polyCorners && count > 0; i++)
+            {
+                int next = i + 1;
+                if (this.closedPath && next == polyCorners)
+                {
+                    next = 0;
+                }
+
+                var point = FindIntersection(this.points[i], this.points[next], start, end);
```

Comment:
`var`... Incidentally the rule used for `var` in the Microsoft libraries is that it can only be used when the type is present on the same line. I don't think many people know this.